v1/ao/Makefile

```diff
@@ -21,3 +25,44 @@ $(cov_merge): test $(cov_files)
 
 coverhtml: $(cov_merge)
 	go tool cover -html=$(cov_merge)
+
+are-you-sure:
+	@read -p "I suppose you know what you are doing. Are you sure? [Y/n]" -n 1 -r; \
+	if [[ $$REPLY != "Y" ]]; then \
+		echo -e -n "\nSee you. >> "; \
+		exit 1; \
+	fi
+	@echo -e "\nGood, proceeding..."; \
+		$(eval confirmed := "Yes")
+
+# Change the value of initVersion and tag it with the version number.
+# Usage: make VERSION=1.2.3 tag-and-release
+tag-and-release: are-you-sure
+	# VERSION must be set
+	@if [ -z "$$VERSION" ]; then \
+		echo -e "The VERSION is unset or an empty string"; \
+		exit 1; \
+	fi
+	# Let you know what you are doing.
+	@echo -e "Changing the version number and tag it with ${VERSION}."
+	# Stash your current work, in case of any
+	@git stash
+	# We tag and release on the master branch
+	@git checkout master
+	# Make sure it's up-to-date
+	@git fetch origin
+	@git reset --hard origin/master
+	# Replace the version with the value provided by you
+	# Please note that this does not work with sed on a Mac.
+	# Install the GNU sed with `brew install gnu-sed --with-default-names` first
```

Comment:
Does something terrible happen with the default `sed` on a Mac, or just that nothing is replaced?  Wondering if we want a further prompt to show the user what's changed before proceeding with the commit + push.

Reply:
The default `sed` of macOS is _BSD sed_ which is incompatible with _GNU sed_ in argument syntax, and the in place flag `-i` in the code does not work the same way with BSD sed. 

A quick Google brings me to this link: https://stackoverflow.com/questions/5694228/sed-in-place-flag-that-works-both-on-mac-bsd-and-linux and I think this line can be improved a little bit then.

Reply:
Added a new commit to support both sed styles(basically macOS(BSD) and Linux(GNU)): https://github.com/appoptics/appoptics-apm-go/pull/28/commits/c952462f3bab40ac7d49a59a8b1bda2c92dad50c

Reply:
👍